python/grass/jupyter/baseseriesmap.py

```diff
@@ -0,0 +1,176 @@
+        slider_width is passed to ipywidgets in ipywidgets.Layout(width=slider_width).
+        """
+        # Lazy Imports
+        import ipywidgets as widgets  # pylint: disable=import-outside-toplevel
+
+        # Render images if they have not been already
+        if not self._layers_rendered:
+            self.render()
+
+        # Set default slider width
+        if not slider_width:
+            slider_width = "70%"
+
+        # Datetime selection slider
+        slider = widgets.SelectionSlider(
+            options,
+            value,
+            description,
+            disabled=False,
+            continuous_update=True,
+            orientation="horizontal",
+            readout=True,
+            layout=widgets.Layout(width=slider_width),
+        )
+        play = widgets.Play(
+            interval=500,
+            value=0,
+            min=0,
+            max=max,
+            step=1,
+            description="Press play",
+            disabled=False,
+        )
+        out_img = widgets.Image(value=b"", format="png")
+
+        def change_slider(change):
+            if isinstance(slider.options[0], tuple):
+                slider.value = slider.options[change.new][1]
+            else:
+                slider.value = slider.options[change.new]
+
+        play.observe(change_slider, names="value")
+
+        # Display image associated with datetime
+        def change_image(date):
+            # Look up layer name for date
+            filename = self._date_filename_dict[date]
+            with open(filename, "rb") as rfile:
+                out_img.value = rfile.read()
+
+        # Return interact widget with image and slider
+        widgets.interactive_output(change_image, {label: slider})
+        layout = widgets.Layout(
+            width="100%", display="inline-flex", flex_flow="row wrap"
+        )
+        return widgets.HBox([play, slider, out_img], layout=layout)
+
+    def save(
+        self,
+        filename,
+        duration=500,
+        label=True,
+        font=None,
+        text_size=12,
+        text_color="gray",
+    ):
+        """
+        Creates a GIF animation of rendered layers.
+
+        Text color must be in a format accepted by PIL ImageColor module. For supported
+        formats, visit:
+        https://pillow.readthedocs.io/en/stable/reference/ImageColor.html#color-names
+
+        param str filename: name of output GIF file
+        param int duration: time to display each frame; milliseconds
+        param bool label: include date/time stamp on each frame
+        param str font: font file
+        param int text_size: size of date/time text
+        param str text_color: color to use for the text.
+        """
+
+        # Render images if they have not been already
+        if not self._layers_rendered:
+            self.render()
+
+        input_files = []
+        for date in self._dates:
+            input_files.append(self._date_filename_dict[date])
```

Comment:
This probably won't work, `self._date_filename_dict` is not in SeriesMap.

Reply:
I realize I may have overlooked this point. Do you think we could base our condition on `_layer_filename_dict`? Since it only exists in seriesmap, we can proceed with a condition: if `_layer_filename_dict is None`, we can follow the method outlined in timeseries map; otherwise, we can continue with the current approach in seriesmap.

Reply:
I think `_layer_filename_dict` and `_layer_filename_dict` are basically the same thing, just the keys are different (integer index in SeriesMap and date in TimeSeriesMap). So use the same variable name for both classes (`_layer_filename_dict`). You could then generalize the `_dates` as e.g. `_index` with the goal of using the same variables in both classes. In SeriesMap the `_index` would be integers.

There are other ways to deal with this, e.g., implement a method `_get_filenames()` in each derived class and call it from the base class. This is probably easier, but not as clean solution I think.

In any case I would encourage you to review Python inheritance concepts.